### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by Allda**
This pull request adds Github actions with the following steps:

- Run tox code checks
-- pytest, black, pylint, mypy
- Build a python package
- Release a package to PyPi
-- each commit/PR pushes package to test instance of Pypi
-- each tag pushes package to production Pypi

### CODE REVIEW

**Comment by Allda** [comment]
@kusha Please review

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+on: push
+
+jobs:
+  tox-test:
+    name: Tox test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Run Tox tests
```
Comment (kusha) [inline .github/workflows/main.yml:13]: Is it beneficial to run it on Fedora? Pulls of `fedorapython/fedora-python-tox` are taking around 25s.

**Review by kusha — APPROVED** [review]
LGTM

File: .github/workflows/main.yml (hunk 1 above)
Comment (Allda) [reply .github/workflows/main.yml:13]: This image has a pre-installed tox. Do you have any alternative?

**Pull request merged**